Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,334 @@
+A REST exporter MAY also inline those links if it can ensure that no name collisions occur.  
+This document proposes that [RFC 6570 URI Template](https://tools.ietf.org/html/rfc6570) SHOULD be used to encode those links.
+
+The following chapters explain how such relationship information is generated.
+
+## 1. Using the path "structure"
+
+The path structure of the Webservice can be used to infer relationship information.  
+For example the fact that the `/user/:userid/post` path is a subroute to `/user/:userid` path can be used to 
+generate relationship information for every user, namely a relationship pointing to `user/{userid}/post`.
+
+In general every endpoint will relate to **all** of its children/subroutes.
+
+Giving an example: When a client sends a request for `/user/532` the a potential REST exporter will return the following:
+```json
+{
+  "id": "532",
+  "name": "Rudi",
+  "lastname": "Raser",
+  "email": "rudi@raser.de",
+  "_links": {
+    "self": "https://example.api/user/532",
+    "post": "https://example.api/user/532/post"
+  }
+}
+```
+
+One of the core concepts of REST (with HATEOAS) is that a client does not need to know anything about the path structure.
+Instead, just like a web browser, a client follows hyperlinks returned from the web service.  
+This allows you to change the path structure of your REST service - keeping the same named links at the same places -
+without breaking compatibility for your clients.
+
+To support such scenarios there must be some sort of way to manually overwrite such relationship information.
+
+## 2. Manually defining "structural" relationship information
+
+This chapter imagines ways how a user could manually override or provide additional relationship information.
+
+### 2.1. Overriding the relationship name
+
+The above described inferring approach will use the string description of the `PathComponent` as the relationship name.
+This may not fit everybody's needs.
+
+Similar to how you can manually define a `@Parameter` name, you can also override a relationship name.
+This proposal suggest a sort of `PathComponentModifier` (similar to `ComponentModifiers`).
+
+Such a definition could then look like the following:
+```swift
+var content: some Component {
+  Group("test".relationshipName("new-name")) {
+    Handler()
+  }
+}
+```
+
+### 2.2. Adding new relationships
+
+In order to support rearranging components, a user can also define its own relationships.
+We introduce a `.relationship` modifier for `PathComponent`s, to which one would pass
+a relationship name and `PathComponent`s describing the destination.  
+The destination can of course also include `@PathParameters`.
+
+Any manual definition of a relationship MUST always override an inferred relationship if the name collides.
+
+Looking at the example below, a REST exporter would add the link `greeter` pointing to `/greeting/{userid}` on a 
```

Comment:
Are you intentionally using two notations for path parameters? `../{param}` and `../:param`

Reply:
Yes, `./{param}` indicates that this parameter is actually replaced by the exporter as it is known at this point. Used the URI Template format here see [RFC 6570 3.2.2 Simple String Expansion: {var}](https://tools.ietf.org/html/rfc6570#section-3.2.2).

Reply:
thanks for the clarification 👍